backend/app/graphql/all_users.py

```diff
@@ -0,0 +1,46 @@
+import graphene
+from ..graphql.services import services
+from .types import QueryList, Query
+
+
+class User(Query):
+    name = graphene.String()
+    email = graphene.String()
+    role = graphene.String()
+
+
+class UserQueries(QueryList):
+    users = graphene.List(
+        User,
+        first=graphene.Int(default_value=5),
+        offset=graphene.Int(default_value=0),
+        role=graphene.String(default_value=""),
+    )
+
+    def resolve_users(self, info, first, offset, role):
+        user_service = services["user_service"]
+        users = user_service.get_users()
+
+        if role != "":
+            filtered = []
+            for user in users:
+                if user.role == role:
+                    filtered.append(
+                        User(
+                            name=f"{user.first_name} {user.last_name}",
+                            email=user.email,
+                            role=user.role,
+                        )
+                    )
+            return filtered[offset : offset + first]  # noqa: E203
+
+        return [
+            *map(
```

Comment:
You probably don't need to manually map over the users in this way. Instead, you can return a list of `UserDTO`s directly; `graphene` should automatically convert them to `User`s. In order to map the `name` field, you can add a custom resolver to the user class, like this:

```python
def resolve_name(parent, info):
    return f"{parent.first_name} {parent.last_name}"
```